src/terms/xc.jl

```diff
@@ -116,17 +129,67 @@ end
     # Note: We always have to do this, otherwise we get issues with AD wrt. scaling_factor
     potential .*= term.scaling_factor
 
+    (; E, potential, Vτ)
+end
+
+@views @timing "ene_ops: xc" function ene_ops(term::TermXc, basis::PlaneWaveBasis{T},
+                                              ψ, occupation; ρ, τ=nothing, kwargs...) where {T}
+    E, Vxc, Vτ = xc_potential_real(term, basis, ψ, occupation; ρ, τ)
+
     ops = map(basis.kpoints) do kpt
         if !isnothing(Vτ)
-            [RealSpaceMultiplication(basis, kpt, potential[:, :, :, kpt.spin]),
+            [RealSpaceMultiplication(basis, kpt, Vxc[:, :, :, kpt.spin]),
              DivAgradOperator(basis, kpt, Vτ[:, :, :, kpt.spin])]
         else
-            RealSpaceMultiplication(basis, kpt, potential[:, :, :, kpt.spin])
+            RealSpaceMultiplication(basis, kpt, Vxc[:, :, :, kpt.spin])
         end
     end
     (; E, ops)
 end
 
+@timing "forces: xc" function compute_forces(term::TermXc, basis::PlaneWaveBasis{T},
+                                             ψ, occupation; ρ, τ=nothing, kwargs...) where {T}
+    # the only non-zero force contribution is from the nlcc core charge
+    # early return if nlcc is disabled / no elements have model core charges
+    isnothing(term.ρcore) && return nothing
+
+    _, Vxc_real, _ = xc_potential_real(term, basis, ψ, occupation; ρ, τ)
+    if basis.model.spin_polarization in (:none, :spinless)
+        Vxc_fourier = fft(basis, Vxc_real[:,:,:,1])
+    else
+        Vxc_fourier = fft(basis, mean(Vxc_real, dims=4))
```

Comment:
This is pretty confusing. The factor of 2 is associated with the density, not the potential, can't you move it there?

Reply:
Also isn't this just `total_density(Vxc_real)`  (once the factor 2 has been resolved).